common/src/main/java/com/microsoft/identity/common/internal/controllers/CommandDispatcher.java

```diff
@@ -489,18 +511,21 @@ private static boolean eligibleToCacheException(BaseException exception) {
     /**
      * Get Commandresult from acquiretokenresult
      *
-     * @param baseException
      * @param result
      */
-    private static CommandResult getCommandResultFromTokenResult(BaseException baseException,
-                                                                 @NonNull AcquireTokenResult result, @NonNull String correlationId) {
-        //Token Commands
-        if (result.getSucceeded()) {
+    private static CommandResult getCommandResultFromTokenResult(@NonNull AcquireTokenResult result, @NonNull String correlationId) {
+
+        final ILocalAuthenticationResult acquireTokenResult = result.getLocalAuthenticationResult();
+        if (acquireTokenResult != null
+                && acquireTokenResult.getAccessTokenRecord().shouldRefresh()) {
+            return new CommandResult(CommandResult.ResultStatus.REFRESH,
+                    result.getLocalAuthenticationResult(), correlationId);
+        } else if (result.getSucceeded()) {
```

Comment:
If we `shouldRefresh` does that mean `result.getSucceeded()` is returning `false`? or does it return `true`? and should we check for that value here?

Reply:
yeah but we're not checking there for the succeeded flag....as I read this code it seems to imply that having to refresh does NOT mean success and to me that seems wrong...even if we happen to do a refresh due to this feature, if those requests are still returning an AT back to the caller then I would consider those to be successful too..

Reply:
Maybe to take a step back, I didn't answer the first question, the result.getSucceded() does return true, and this code **Outdated**, the updated version prioritizes the result.getSucceded() before the acquireTokenResult.getAccessTokenRecord().shouldRefresh() in the branch logic.